mungegithub/mungers/lgtm_handler.go

```diff
@@ -70,14 +71,23 @@ func (h LGTMHandler) Munge(obj *github.MungeObject) {
 		return
 	}
 
+	events, err := obj.GetEvents()
+	if err != nil {
+		glog.Errorf("unexpected error getting events: %v", err)
+		return
+	}
+
 	if !obj.HasLabel(lgtmLabel) {
-		h.addLGTMIfCommented(obj, comments, reviewers)
+		h.addLGTMIfCommented(obj, comments, events, reviewers)
 		return
 	}
-	h.removeLGTMIfCancelled(obj, comments, reviewers)
+	h.removeLGTMIfCancelled(obj, comments, events, reviewers)
 }
 
-func (h *LGTMHandler) addLGTMIfCommented(obj *github.MungeObject, comments []*githubapi.IssueComment, reviewers mungerutil.UserSet) {
+func (h *LGTMHandler) addLGTMIfCommented(obj *github.MungeObject, comments []*githubapi.IssueComment, events []*githubapi.IssueEvent, reviewers mungerutil.UserSet) {
+	// Get the last time when the someone applied lgtm manually.
+	removeLGTMTime := e.LastEvent(events, e.And{e.RemoveLabel{}, e.LabelName(lgtmLabel), e.HumanActor()}, nil)
```

Comment:
This has to be the worst function signature in contrib right now. I hope.  If we ever get in this are again lets make this better...  somehow...